File: solutions/csharp/space-age/SpaceAge.cs
using System;
using System.Collections.Generic;

public class SpaceAge
{
    private const double earthYearInSeconds = 31557600.0;
    private enum Planet { Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, Neptune }
    private readonly Dictionary<Planet, double> orbitalPeriod = new Dictionary<Planet, double>()
    {
        {Planet.Mercury, 0.2408467},
        {Planet.Venus,   0.61519726},
        {Planet.Earth,   1.0},
        {Planet.Mars,    1.8808158},
        {Planet.Jupiter, 11.862615},
        {Planet.Saturn,  29.447498},
        {Planet.Uranus,  84.016846},
        {Planet.Neptune, 164.79132},
    };

    private readonly double yearsOnEarth;

    public SpaceAge(int seconds) => yearsOnEarth = seconds / earthYearInSeconds;

    public double OnEarth() => yearsOnEarth / orbitalPeriod[Planet.Earth];
    public double OnMercury() => yearsOnEarth / orbitalPeriod[Planet.Mercury];
    public double OnVenus() => yearsOnEarth / orbitalPeriod[Planet.Venus];
    public double OnMars() => yearsOnEarth / orbitalPeriod[Planet.Mars];
    public double OnJupiter() => yearsOnEarth / orbitalPeriod[Planet.Jupiter];
    public double OnSaturn() => yearsOnEarth / orbitalPeriod[Planet.Saturn];
    public double OnUranus() => yearsOnEarth / orbitalPeriod[Planet.Uranus];
    public double OnNeptune() => yearsOnEarth / orbitalPeriod[Planet.Neptune];
}

File: solutions/ruby/space-age/space_age.rb
class SpaceAge
  EARTH_YEAR_IN_SECONDS = 31557600.freeze
  ORBITAL_PERIOD = {
    mercury: 0.2408467,
    venus: 0.61519726,
    earth: 1.0,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132,
  }.freeze

  def initialize(seconds)
    @earth_age = seconds / EARTH_YEAR_IN_SECONDS.to_f
  end

  ORBITAL_PERIOD.each do |planet, planetary_period|
    define_method "on_#{planet}" do
      @earth_age / planetary_period
    end
  end

  private_constant :ORBITAL_PERIOD
  private_constant :EARTH_YEAR_IN_SECONDS
end
